src/js/binaryen.js-post.js

```diff
@@ -2158,6 +2158,28 @@ function wrapModule(module, self) {
       );
     });
   };
+  self['getFunctionTable'] = function() {
+    return {
+      'imported': Boolean(Module['_BinaryenIsFunctionTableImported'](module)),
+      'segments': (function() {
+        var arr = [];
+        for (var i = 0, j = Module['_BinaryenGetNumFunctionTableSegments'](module) ; i !== j ; ++i)
+        {
+          var seg = {'offset': Module['_BinaryenGetFunctionTableSegmentOffset'](module, i), 'names': []};
+          for (var k = 0, l = Module['_BinaryenGetFunctionTableSegmentLength'](module, i) ; k !== l ; ++k)
+          {
+            var size = Module['_BinaryenGetFunctionTableSegmentDataByteLength'](module, i, k);
+            var ptr = _malloc(size);
+            Module['_BinaryenCopyFunctionTableSegmentData'](module, i, k, ptr);
```

Comment:
I looked through binaryen.js-post.js to see how strings are usually copied without calling `malloc`, and it looks like all the functions that return strings just return a `char *` pointer to the original string rather than copying the string contents as if it were an arbitrary byte buffer.

Following that pattern, we could either have `BinaryenGetFunctionTableSegmentData` take both a segment index and a data index and have it return a `char *` or we could have `BinaryenGetFunctionTableSegmentData` copy all the data `char *`s into a buffer (which is closer to what you had before). @kripken, do you have an opinion on which would be better?

Reply:
@kripken ping!

Reply:
Sorry for missing this!

Is this getting the names of the functions in the table? We can assume they stay alive while we translate the C name into a JS name, so I don't think malloc is needed. Yeah, it can return a `char*` to the name from C, I think.

Reply:
I talked with @kripken more about this offline, and it sounds like we should start out with the API where you pass it an index and it returns just one `char *`.  That way the caller doesn't need to worry about allocating any buffers at all. Does that sound good, @COFFEETALES?